src/eth_provider/provider.rs

```diff
@@ -1004,6 +1011,8 @@ where
                     .await?
                     .is_some()
             {
+                tracing::info!("Pruning pending transaction: {hash:?} after reaching max retries or already finalized");
```

Comment:
Same, having display implemented would make it nicer I think

Reply:
Yes agreed, fixed